From 981c354ebb0536c869ef68a1671366332d25e5c4 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Wed, 14 Sep 2016 17:14:33 +0200
Subject: network: Fix crash entering IPv4 routes

The wrong destructor was used for a GPtrArray of NMIPRoute.

See https://bugzilla.gnome.org/show_bug.cgi?id=771251
See https://bugzilla.gnome.org/show_bug.cgi?id=771432
---
 panels/network/connection-editor/ce-page-ip4.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/panels/network/connection-editor/ce-page-ip4.c b/panels/network/connection-editor/ce-page-ip4.c
index 70187e8..51995f9 100644
--- a/panels/network/connection-editor/ce-page-ip4.c
+++ b/panels/network/connection-editor/ce-page-ip4.c
@@ -794,7 +794,7 @@ ui_to_setting (CEPageIP4 *page)
                 g_ptr_array_add (dns_servers, NULL);
         }
 
-        routes = g_ptr_array_new_with_free_func (g_object_unref);
+        routes = g_ptr_array_new_with_free_func (nm_ip_route_unref);
         if (g_str_equal (method, NM_SETTING_IP4_CONFIG_METHOD_AUTO) ||
             g_str_equal (method, NM_SETTING_IP4_CONFIG_METHOD_MANUAL))
                 children = gtk_container_get_children (GTK_CONTAINER (page->routes_list));
-- 
cgit v0.12

